datalad/core/local/tests/test_save.py

```diff
@@ -208,6 +208,38 @@ def test_subdataset_save(path):
     assert_repo_status(parent.path, untracked=['untracked'], modified=['sub'])
 
 
+@with_tempfile(mkdir=True)
+def test_subsuperdataset_save(path):
+    # Verify that when invoked without recursion save does not
+    # cause querying of subdatasets of the subdataset
+    # see https://github.com/datalad/datalad/issues/4523
+    parent = Dataset(path).create()
+    # Create 3 levels of subdatasets so later to check operation
+    # with or without --dataset being specified
+    sub1 = parent.create('sub1')
+    sub2 = parent.create(sub1.pathobj / 'sub2')
+    sub3 = parent.create(sub2.pathobj / 'sub3')
+    assert_repo_status(path)
+    # now we will lobotomize that sub2 so git would fail if any query is performed.
+    rmtree(str(sub3.pathobj / '.git' / 'objects'))
+    # the call should proceed fine since neither should care about sub3
+    # default is no recursion
+    parent.save('sub1')
+    sub1.save('sub2')
+    # and should fail if we demand recursive operation
+    # Fun part: causes RecursionError, not just CommandError ATM but that is IMHO
+    # a separate issue, TODO.
```

Comment:
Isn't this comment stale as of d5d967015?

Reply:
d'oh right -- will remove.  It seems that everyone agrees, so I will remove, commit, merge locally and push master to avoid needless round of CI